spec/polaris-management-service.yml

```diff
@@ -1248,6 +1283,20 @@ components:
         - currentEntityVersion
         - properties
 
+    ResetPrincipalRequest:
+      type: object
+      properties:
+        clientId:
+          type: string
+          description: >
+            Optional client ID to set for the principal.
+            Must be a valid client ID previously generated by this service.
+        clientSecret:
+          type: string
+          description: >
+            Optional client secret to set for the principal.
+            Must match the secret issued for the given client ID.
```

Comment:
note sure about the "must match" part... How about "Polaris service implementations may impose extra requirements on what is accepted as a secret (special chars, length, etc.)".